src/main/java/teammates/logic/core/AccountsLogic.java

```diff
@@ -238,4 +242,42 @@ private void createStudentAccount(StudentAttributes student)
         accountsDb.createEntity(account);
     }
 
+    /**
+     * Updates the read notifications of an account.
+     */
+    public AccountAttributes updateReadNotifications(String googleId, String notificationId, Instant endTime)
+            throws InvalidParametersException, EntityDoesNotExistException {
+        AccountAttributes a = accountsDb.getAccount(googleId);
+        if (a == null) {
+            throw new EntityDoesNotExistException("Invalid account id.");
+        }
+
+        if (endTime.isBefore(Instant.now())) {
+            throw new InvalidParametersException("Notification has expired.");
+        }
+        Map<String, Instant> updatedReadNotifications = a.getReadNotifications();
+
+        for (Map.Entry<String, Instant> notification : updatedReadNotifications.entrySet()) {
+            if (notification.getValue().isBefore(Instant.now())) {
+                updatedReadNotifications.remove(notification.getKey());
+            }
+        }
+
+        if (!notificationsLogic.doesNotificationExists(notificationId)) {
+            throw new EntityDoesNotExistException("Invalid notification id to read");
+        }
+
+        updatedReadNotifications.put(notificationId, endTime);
+
+        try {
+            return accountsDb.updateAccount(
+                    AccountAttributes.updateOptionsBuilder(googleId)
+                            .withReadNotifications(updatedReadNotifications)
+                            .build());
+        } catch (InvalidParametersException e) {
+            assert false : "Invalid account data detected unexpectedly "
+                    + "while updating read notifications for account " + googleId + ": " + e.getMessage();
+        }
+        throw new EntityDoesNotExistException("Invalid update to read notification in account");
+    }
```

Comment:
Is this necessary? Since there were already checks for existence before.

Reply:
This line exists as a return statement is needed for the method.